### DISCUSSION TITLE
Can we rely on headers for amqp notifications?

### DISCUSSION THREAD

**Discussion by julb** [minio/minio#16119] [Q&A] [answered]
Hello all!

I noticed that, for amqp event notifications, minio sets the bucket name as an header in the message. That's definitely useful for my use case.

I have browsed the documentation and there is no mention of it. Is it something that we can rely on for future versions of minio ?

Thanks !

**Comment by klauspost** [comment, marked as answer]
I don't see why that would change, but I also cannot give any guarantees.

**Comment by julb** [comment]
Thanks for your feedback.
My use case was that we could use an Headers exchange type in AMQP with these headers.
It would be helpful then for multiple buckets to use the same AMQP event configuration, and redirect to appropriate queue with native RabbitMQ filtering (`x-match`).
Would be nice to mention it in the docs.
Thanks for your work.

**Reply by harshavardhana** [reply]
We will not change it, it was added on purpose.

// cc @ravindk89 we can mention a one liner on why we have the amqp routing built-in
